src/dune_digest/dune_digest.ml

```diff
@@ -112,43 +116,59 @@ exception
 
 let directory_digest_version = 2
 
-let rec path_with_stats ~allow_dirs path (stats : Stats_for_digest.t) :
+let path_with_stats ~allow_dirs path (stats : Stats_for_digest.t) :
     Path_digest_result.t =
+  let rec loop path (stats : Stats_for_digest.t) =
+    match stats.st_kind with
+    | S_LNK ->
+      let executable =
+        Path.Permissions.test Path.Permissions.execute stats.st_perm
+      in
```

Comment:
You can lift this out of the match and share it with the other branch. Maybe thunk it if you don't want it evaluated?